runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContent.java

```diff
@@ -124,6 +125,17 @@ public void run() {
         }
     }
 
+    private static WindowAndroid getWindowAndroid(Context context) {
+        WindowAndroid wrapper;
+        boolean contextWrapsActivity = activityFromContext(context) != null;
+        if (contextWrapsActivity) {
+            wrapper = new ActivityWindowAndroid(context);
+        } else {
+            wrapper = new WindowAndroid(context);
+        }
+        return wrapper;
+    }
+
```

Comment:
Nice. 
But I don't understand what's the purpose of the code in AwContents.java as below. Is there any special consideration to save the WindowAndroidWrapper? Do we need porting this part totally?

   ```
private static WeakHashMap<Context, WindowAndroidWrapper> sContextWindowMap;
    @SuppressFBWarnings("LI_LAZY_INIT_STATIC")
    private static WindowAndroidWrapper getWindowAndroid(Context context) {
        if (sContextWindowMap == null) sContextWindowMap = new WeakHashMap<>();
        WindowAndroidWrapper wrapper = sContextWindowMap.get(context);
        if (wrapper != null) return wrapper;

        boolean contextWrapsActivity = activityFromContext(context) != null;
        if (contextWrapsActivity) {
            final boolean listenToActivityState = false;
            wrapper = new WindowAndroidWrapper(
                    new ActivityWindowAndroid(context, listenToActivityState));
        } else {
            wrapper = new WindowAndroidWrapper(new WindowAndroid(context));
        }
        sContextWindowMap.put(context, wrapper);
        return wrapper;
    }
```